python-frontend/src/main/java/org/sonar/plugins/python/api/caching/PythonReadCache.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.api.caching;
+
+import java.io.InputStream;
+import javax.annotation.CheckForNull;
+
+public interface PythonReadCache {
+  InputStream read(String var1);
+
+  @CheckForNull
+  byte[] readBytes(String var1);
+
+  boolean contains(String var1);
```

Comment:
It would be nice to have some documentation on these, just as we have them in SonarJava (for example notes on whose responsibility it is to close the stream).